### ISSUE TITLE
CMake 3.30: Calling FetchContent_Populate() is deprecated

### ISSUE THREAD

**Issue by gerhardol** [cpm-cmake/CPM.cmake#568]
New warning with updated CMake 3.30-rc3:

```
CMake Warning (dev) at /usr/local/src/cmake-3.30.0/share/cmake-3.30/Modules/FetchContent.cmake:1942 (message):
  Calling FetchContent_Populate(MyTools) is deprecated, call
  FetchContent_MakeAvailable(MyTools) instead.  Policy CMP0169 can be set
  to OLD to allow FetchContent_Populate(MyTools) to be called directly for
  now, but the ability to call it with declared details will be removed
  completely in a future version.
Call Stack (most recent call first):
  /home/t052809/gu/git_sync/gev/avap_applicationupdater/avap-util/CMakeConfig/cpm/CPM.cmake:1077 (FetchContent_Populate)
  /home/t052809/gu/git_sync/gev/avap_applicationupdater/avap-util/CMakeConfig/cpm/CPM.cmake:849 (cpm_fetch_package)
  CMakeLists.txt:7 (cpmaddpackage)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

**Comment by Curve** [comment]
This discussion https://gitlab.kitware.com/cmake/cmake/-/merge_requests/9552#note_1526701 might be useful

**Comment by robertmaynard** [comment]
The conversion of CPM over to `FetchContent_MakeAvailable` is great but did bring to the surface some underlying issues with lowercase project names when used with SOURCE_SUBDIR. 

The fix has been merged and will be part of 3.30.3 ( background https://gitlab.kitware.com/cmake/cmake/-/issues/26243 ). But I wanted the CPM team to be aware in case you see issues filed about it.
